extra/boost_unordered.natvis

```diff
@@ -0,0 +1,380 @@
+    </Type>
+
+    <Type Name="boost::unordered::detail::foa::atomic_integral&lt;*&gt;" Inheritable="false">
+        <Intrinsic Name="get" Expression="n._Storage._Value" />
+    </Type>
+
+    <Type Name="boost::unordered::detail::foa::group15&lt;*&gt;" Inheritable="false">
+        <Intrinsic Name="check_bit" Expression="(m[b].get() == 0) ? (1 &lt;&lt; b) : 0">
+            <Parameter Name="b" Type="int" />
+        </Intrinsic>
+        <Intrinsic Name="__match_occupied_regular_layout_true" Expression="0x7FFF &amp; ~(
+                check_bit( 0) + check_bit( 1) + check_bit( 2) + check_bit( 3) +
+                check_bit( 4) + check_bit( 5) + check_bit( 6) + check_bit( 7) +
+                check_bit( 8) + check_bit( 9) + check_bit(10) + check_bit(11) +
+                check_bit(12) + check_bit(13) + check_bit(14) + check_bit(15)
+        )" />
+        <Intrinsic Name="__match_occupied_x" Expression="m[0].get() | m[1].get()" />
+        <Intrinsic Name="__match_occupied_y" Expression="static_cast&lt;uint32_t&gt;(__match_occupied_x() | (__match_occupied_x() &gt;&gt; 32))" />
+        <Intrinsic Name="__match_occupied_regular_layout_false" Expression="0x7FFF &amp; (__match_occupied_y() | (__match_occupied_y() &gt;&gt; 16))" />
+        <Intrinsic Name="match_occupied" Expression="regular_layout ? __match_occupied_regular_layout_true() : __match_occupied_regular_layout_false()" />
+
+        <Intrinsic Name="__is_sentinel_regular_layout_true" Expression="pos == N-1 &amp;&amp; m[N-1].get() == sentinel_" >
+            <Parameter Name="pos" Type="size_t" />
+        </Intrinsic>
+        <Intrinsic Name="__is_sentinel_regular_layout_false" Expression="pos == N-1 &amp;&amp; (m[0].get() &amp; uint64_t(0x4000400040004000ull)) == uint64_t(0x4000ull) &amp;&amp; (m[1].get() &amp; uint64_t(0x4000400040004000ull))==0" >
+            <Parameter Name="pos" Type="size_t" />
+        </Intrinsic>
+        <Intrinsic Name="is_sentinel" Expression="regular_layout ? __is_sentinel_regular_layout_true(pos) : __is_sentinel_regular_layout_false(pos)" >
+            <Parameter Name="pos" Type="size_t" />
+        </Intrinsic>
+    </Type>
+
+    <!-- FOA and CFOA stats -->
+
+    <Type Name="boost::unordered::detail::foa::table_core_cumulative_stats" Inheritable="false">
+        <DisplayString>stats</DisplayString>
+        <Expand>
+            <Item Name="[insertion]">insertion</Item>
+            <Item Name="[successful_lookup]">successful_lookup</Item>
+            <Item Name="[unsuccessful_lookup]">unsuccessful_lookup</Item>
+        </Expand>
+    </Type>
+
+    <Type Name="boost::unordered::detail::cumulative_stats&lt;*&gt;" Inheritable="true">
+        <Intrinsic Name="bit_cast_to_double" Expression="*reinterpret_cast&lt;double*&gt;(&amp;i)">
+            <Parameter Name="i" Type="size_t" />
+        </Intrinsic>
+        <Intrinsic Name="bit_cast_to_size_t" Expression="*reinterpret_cast&lt;size_t*&gt;(&amp;d)">
```

Comment:
This crashes in x86 mode:

![imagen](https://github.com/boostorg/unordered/assets/3283164/d759f936-41b2-4b9d-9ded-8b1d073af51a)

I could fix it locally by rewriting as follows:
```
        <Intrinsic Name="bit_cast_to_double" Expression="*reinterpret_cast&lt;double*&gt;(&amp;i)">
            <Parameter Name="i" Type="uint64_t" />
        </Intrinsic>
        <Intrinsic Name="bit_cast_to_uint64_t" Expression="*reinterpret_cast&lt;uint64_t*&gt;(&amp;d)">
            <Parameter Name="d" Type="double" />
        </Intrinsic>
        
        <!-- https://en.wikipedia.org/wiki/Fast_inverse_square_root#Magic_number -->
        <Intrinsic Name="__inv_sqrt_init" Expression="bit_cast_to_double(0x5FE6EB50C7B537A9ull - (bit_cast_to_uint64_t(x) &gt;&gt; 1))">
            <Parameter Name="x" Type="double" />
        </Intrinsic>
```

(FWIW, one of the changes (s/0x5FE6EB50C7B537A9/0x5FE6EB50C7B537A9ul) is probably not necessary, but felt it look cleaner.)

Reply:
Thanks, I'll make those changes.